### ISSUE TITLE
常用模块类型

### ISSUE THREAD

**Issue by achangzx** [achangzx/machine-learning#1]

**Comment by achangzx** [comment]
torch中的utils中的tensorboard中的SummaryWriter是用来对数据进行可视化的
导入方式为：from torch.utils.tensorboard import SummaryWriter
a=SummaryWriter（‘传入目录’）
a.add_images    仅仅可以传入tensor或者numpy型的图片
a.add_scalar       输入自己绘制的图像

**Comment by achangzx** [comment]
torchvision中的transforms可以用来转换图片的格式为tensor类型
导入方式为：from torchvision import transforms
fangshi=transforms.ToTensor(传入被Image打开的原图片)
PS：PIL中的Image是用来处理图片的一个模块
导入方式为：from PIL import Image
c=Image.open('图片的绝对路径或者相对路径')

**Comment by achangzx** [comment]
os模块可以将我们的图片输出为一个列表的形式
导入方式：import os
path=os.path.join('根目录','下一层目录')
list=os.listdir(path)

**Comment by achangzx** [comment]
批量处理图片数据要用到torchvision中的datasets可以将图片组下载到我们需要的位置
导入方式：from torchvision import datasets
test=torchvision.datasets.(目标图片组)（）
批量加载处理图片用的是torch中的utils中的data中的DataLoader
导入方式为from torch.utils,data import DataLoader
test_loader=DataLoader()
实例：
import torchvision.datasets
from torch.utils.tensorboard import SummaryWriter

test_datasets=torchvision.datasets.CIFAR10('./批量处理数据',train=False,transform=torchvision.transforms.ToTensor(),download=True)
from torch.utils.data import DataLoader
test_data_loader=DataLoader(dataset=test_datasets,batch_size=60,shuffle=True,drop_last=False)
#数据加载方法的作用是把一组数据，每10个分为一组，且是随机抓取，并且保留最后不足10的那部分
print(test_datasets.classes)
print(type(test_data_loader))
print(len(test_data_loader))
#for 循环本质是对一个列表进行迭代，常用的即为range(10)就是0到9的一个列表
#处理数据可视化的工具即为SummaryWriter
#a=SummaryWriter('输入目录地址')
#a.add_images()填参数即可
#打开终端tensorboard --logdir='目录地址'
a=SummaryWriter('dataloader')
for data in test_data_loader:
    images,targets=data
    print(images.shape)
    print(targets)
    print('\n')
step=0
for data2 in test_data_loader:
    images2,targets=data2
    a.add_images('test_data',img_tensor=images2,global_step=step)
    step=step+1
a.close()
